File: LeetCode/Bit/JS/Leetcode2044-统计按位或能得到最大值的子集数目.js
var countMaxOrSubsets = function (nums) {
    const n = nums.length;
    let mxor = 0, ans = 0;
    for (let i = 1; i < 1 << n; i++) {
        let cur = 0
        for (let j = 0; j < n; j++) {
            if (i >> j & 1) cur |= nums[j];
        }
        if (cur == mxor) {
            ans++;
        } else if (cur > mxor) {
            ans = 1;
            mxor = cur;
        }
    }
    return ans;
};

File: LeetCode/Bit/Golang/Leetcode2044-统计按位或能得到最大值的子集数目.go
package sbgo

func countMaxOrSubsets(nums []int) int {
	n, mxor, ans := len(nums), 0, 0
	for i := 1; i < 1<<n; i++ {
		cur := 0
		for j := 0; j < n; j++ {
			if i>>j&1 == 1 {
				cur |= nums[j]
			}
		}
		if cur == mxor {
			ans++
		} else if cur > mxor {
			mxor = cur
			ans = 1
		}
	}
	return ans
}
